### PR TITLE
ansible-navigator: init at 24.2.0; ansible-builder: init at 3.0.1; bindep: init at 2.11.0; onigurumacffi: init at 1.3.0

### PR DESCRIPTION
**Pull request by Melkor333**
## Description of changes

Initialize `ansible-navigator`. This also required me to initialize 3 other dependencies.
Please just close this if I should create separate PR's for the packages.

I'm not very deep into Python packaging, so there might be mistakes.

I'd prefer not being maintainer of the 2 dependencies, therefore I left the maintainers field empty. If that's not OK, I can add myself as Maintainer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Atemu** [comment]
@ofBorg eval

File: pkgs/development/python-modules/ansible-navigator/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    podman
+    pkgs.ansible-lint
+    jinja2
+    jsonschema
+    tzdata
+    onigurumacffi
+  ];
+
+  patchPhase = ''
+    # upper version limit on setuptools and setuptools-scm-git-archive aren't required
```
Comment (Atemu) [inline pkgs/development/python-modules/ansible-navigator/default.nix:36]: Please create an upstream issue (or ideally a patch) on the latter.

File: pkgs/development/python-modules/bindep/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  propagatedBuildInputs = with python3Packages; [
+    parsley
+    pbr
+    packaging
+    distro
+  ];
+
+  patchPhase = ''
+    export PBR_VERSION=5.11.1
```
Comment (Atemu) [inline pkgs/development/python-modules/bindep/default.nix:29]: Explain this magic env var using a code comment.

File: pkgs/development/python-modules/bindep/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Bindep is a tool for checking the presence of binary packages needed to use an application / library";
+    homepage = "https://docs.opendev.org/opendev/bindep/latest/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
```
Comment (Atemu) [inline pkgs/development/python-modules/bindep/default.nix:36]: You're not going to maintain this? That's a bit :/

File: pkgs/development/python-modules/onigurumacffi/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  version = "1.3.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-d0XNxWCWrOyIofOwhmCiKwnGWe040/WdtsHK12qXa+8=";
+  };
+
+  buildInputs = with python3Packages; [
+    oniguruma.dev
```
Comment (Atemu) [inline pkgs/development/python-modules/onigurumacffi/default.nix:13]: buildInputs automatically selects the dev output.

```suggestion
    oniguruma
```

File: pkgs/development/python-modules/onigurumacffi/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    setuptools
+    cffi
+  ];
+
+  meta = with lib; {
+    description = "Python cffi bindings for the oniguruma regex engine";
+    homepage = "https://github.com/asottile/onigurumacffi";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (Atemu) [inline pkgs/development/python-modules/onigurumacffi/default.nix:22]: :/

File: pkgs/development/python-modules/ansible-builder/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  propagatedBuildInputs = with python3Packages; [
+    podman
+    jsonschema
+    requirements-parser
+    pyyaml
+  ];
+
+  patchPhase = ''
+    # the upper limits of setuptools are unnecessary
+    sed -i 's/, <=[0-9.]*//g' pyproject.toml
```
Comment (Atemu) [inline pkgs/development/python-modules/ansible-builder/default.nix:31]: I haven't kept up with python packaging but doesn't the relax deps hook support pyproject.toml by now? If so, please use that instead.

**Review by Atemu — CHANGES_REQUESTED** [review]
Please split your commits into one for each logical unit of work (in this case: one for each package; deps first)

File: pkgs/development/python-modules/ansible-navigator/default.nix (hunk 1 above)
Comment (Melkor333) [reply pkgs/development/python-modules/ansible-navigator/default.nix:36]: The upper version limit was actually a mistake, it's only needed for `ansible-builder`. The git-archive thing is resolved upstream but not release yet. I hope it's fine like that

File: pkgs/development/python-modules/ansible-builder/default.nix (hunk 6 above)
Comment (Melkor333) [reply pkgs/development/python-modules/ansible-builder/default.nix:31]: Sadly it doesn't work with the relax deps hook. But I added a link to the upstream ticket to resolve this issue.

**Comment by Melkor333** [comment]
@Atemu btw. thank you very very much for your review! Given my nixpkgs PR experience, I expected to wait a few months 😄

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 290631` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>ansible-builder (python311Packages.ansible-builder)</li>
    <li>ansible-builder.dist (python311Packages.ansible-builder.dist)</li>
    <li>ansible-navigator (python311Packages.ansible-navigator)</li>
    <li>ansible-navigator.dist (python311Packages.ansible-navigator.dist)</li>
    <li>python311Packages.bindep</li>
    <li>python311Packages.bindep.dist</li>
    <li>python311Packages.onigurumacffi</li>
    <li>python311Packages.onigurumacffi.dist</li>
  </ul>
</details>

**Pull request merged**